encode.go

```diff
@@ -116,7 +126,21 @@ func marshalSlice(field reflect.Value) []string {
 	return out
 }
 
+func compatibleInterfaceValue(field reflect.Value) (string, bool) {
```

Comment:
In the end I think relying on `encoding.TextMarshaler` and `fmt.Stringer` interfaces for compatible types is a mistake, as using their methods implicitly might be a problem for structs that have a different logic for them than what qstring would expect.

I think a better idea would be to create a `qstring.Marshaler` interface which types would be required to implement explicitly. 

A concrete example would be a named type for a slice. A qstring specific implementation would have to prepend the property name (or qstring tag) for every value, but a plain Stringer/TextMarshaler method does not have the complete information for it, and as such would return a broken value.